File: Domain/Entities/Curriculum/SyllabusElements/Course.cs
namespace Domain.Entities.Curriculum.SyllabusElements;

/// <summary>
/// A tree of lessons (with multiple heads)
/// </summary>
public class Course
{
    public int Id { get; set; }
    public string Name { get; set; } = string.Empty;
    public Science? Science { get; set; }
    
    private readonly HashSet<Lesson> startingLessons = new();

    /// <summary>
    /// Courses may have multiple starting points (lessons) - this is a hashset of those.
    /// </summary>
    public HashSet<Lesson> StartingLessons => startingLessons;

    public void AddStartingLesson(Lesson lesson)
    {
        startingLessons.Add(lesson);
    }
    public void RemoveStartingLesson(Lesson lesson)
    {
        startingLessons.Remove(lesson);
    }
}

File: Domain/Entities/Curriculum/SyllabusElements/Course.java
import java.util.HashSet;

public class Course {

    private int id;
    private String name = "";
    private Science science;
    private HashSet<Lesson> startingLessons = new HashSet<Lesson>();
     public int getId() {
         return id;
     }

     public void setId(int id) {
         this.id = id;
     }

     public String getName() {
         return name;
     }

     public void setName(String name) {
         this.name = name;
     }

     public Science getScience() {
         return science;
     }

     public void setScience(Science science) {
         this.science = science;
     }

     public HashSet<Lesson> getStartingLessons() {
         return startingLessons;
     }

     public void addStartingLesson(Lesson lesson) {
         startingLessons.add(lesson);
     }

     public void removeStartingLesson(Lesson lesson) {
         startingLessons.remove(lesson);
     }


}
